docs/aws/audit/esmonitoring/rules/elasticache_redis_cluster_automatic_backup.mdx

```diff
@@ -23,6 +23,206 @@ CBP,SEBI,RBI_MD_ITF,RBI_UCB
+3. In the ElastiCache dashboard, select "Redis" from the left-hand menu. This will bring up a list of all your Redis clusters.
+4. For each cluster, check the "Backup" section. If the "Automatic Backup" field is set to "Disabled", then the cluster does not have automatic backup enabled.
+</Accordion>
+
+<Accordion title='Using CLI'>
+1. Install and configure AWS CLI: Before you can start using AWS CLI, you need to install it on your local system. You can download it from the official AWS website. After installation, you need to configure it with your AWS account credentials. You can do this by running the command `aws configure` and then entering your AWS Access Key ID, Secret Access Key, Default region name, and Default output format when prompted.
+
+2. List all Elasticache Clusters: Use the following AWS CLI command to list all the Elasticache clusters in your AWS account.
+
+   ```
+   aws elasticache describe-cache-clusters --query 'CacheClusters[*].[CacheClusterId]'
+   ```
+
+   This command will return a list of all the Elasticache cluster IDs.
+
+3. Check Automatic Backup for each Elasticache Cluster: For each Elasticache cluster ID returned by the previous command, use the following AWS CLI command to check if automatic backup is enabled.
+
+   ```
+   aws elasticache describe-cache-clusters --cache-cluster-id <CacheClusterId> --query 'CacheClusters[*].[SnapshotRetentionLimit]'
+   ```
+
+   Replace `<CacheClusterId>` with the ID of the Elasticache cluster you want to check. This command will return the number of days for which ElastiCache will retain automatic snapshots. If the returned value is 0, it means that automatic backup is not enabled for this cluster.
+
+4. Analyze the output: If the `SnapshotRetentionLimit` for any of the Elasticache clusters is 0, it means that automatic backup is not enabled for that cluster. This is a misconfiguration and should be fixed.
+</Accordion>
+
+<Accordion title='Using Python'>
+1. Import the necessary AWS SDK for Python (Boto3) modules and establish a session with AWS using your credentials.
+
+```python
+import boto3
+from botocore.exceptions import BotoCoreError, ClientError
+
+session = boto3.Session(
+    aws_access_key_id='YOUR_ACCESS_KEY',
+    aws_secret_access_key='YOUR_SECRET_KEY',
+    region_name='us-west-2'
+)
+```
+
+2. Create a client for 'elasticache' service and retrieve all the Elasticache clusters.
+
+```python
+elasticache = session.client('elasticache')
+
+try:
+    response = elasticache.describe_cache_clusters()
+except ClientError as e:
+    print(f"Unexpected error: {e}")
+```
+
+3. Iterate over each Elasticache cluster and check the 'SnapshotRetentionLimit' attribute. If the value is 0, it means automatic backup is not enabled for that cluster.
+
+```python
+for cluster in response['CacheClusters']:
+    if cluster['SnapshotRetentionLimit'] == 0:
+        print(f"Automatic backup is not enabled for the cluster: {cluster['CacheClusterId']}")
+```
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>